5/Dockerfile

```diff
@@ -1,49 +1,41 @@
-    mkdir -p /etc/apt/keyrings && \
-    curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg && \
-    echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODE_MAJOR}.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list && \
-    apt-get -qqy update && \
-    apt-get --no-install-recommends -qqy install git unzip wget bash fonts-dejavu python3 python3-pip shellcheck nodejs build-essential && \
-    curl -fsSL -o /opt/sonar-scanner-cli.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_SCANNER_VERSION}.zip && \
-    curl -fsSL -o /opt/sonar-scanner-cli.zip.asc https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_SCANNER_VERSION}.zip.asc && \
+RUN set -eux; \
+    addgroup -S -g 1000 scanner-cli; \
+    adduser -S -D -u 1000 -G scanner-cli scanner-cli; \
+    apk add --no-cache --virtual build-dependencies wget unzip gnupg; \
+    apk add --no-cache git python3 py3-pip py3-pylint bash shellcheck "nodejs>=18" openjdk17-jre curl musl-locales musl-locales-lang; \
+    wget -U "scannercli" -q -O /opt/sonar-scanner-cli.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_SCANNER_VERSION}.zip; \
+    wget -U "scannercli" -q -O /opt/sonar-scanner-cli.zip.asc https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_SCANNER_VERSION}.zip.asc; \
     for server in $(shuf -e hkps://keys.openpgp.org \
                             hkps://keyserver.ubuntu.com) ; do \
         gpg --batch --keyserver "${server}" --recv-keys 679F1EE92B19609DE816FDE81DB198F93525EC1A && break || : ; \
-    done && \
-    gpg --verify /opt/sonar-scanner-cli.zip.asc /opt/sonar-scanner-cli.zip && \
-    unzip sonar-scanner-cli.zip && \
-    rm sonar-scanner-cli.zip sonar-scanner-cli.zip.asc && \
-    mv sonar-scanner-${SONAR_SCANNER_VERSION} ${SONAR_SCANNER_HOME} && \
-    pip install --no-cache-dir --upgrade pip && \
-    pip install --no-cache-dir pylint && \
-    mkdir -p "${SRC_PATH}" "${SONAR_USER_HOME}" "${SONAR_USER_HOME}/cache" && \
-    chmod -R 555 "${SONAR_SCANNER_HOME}" && \
-    chmod -R 777 "${SRC_PATH}" "${SONAR_USER_HOME}" "/opt/java/openjdk/lib/security/cacerts" && \
-    rm -rf /var/lib/apt/lists/*
+    done; \
+    gpg --verify /opt/sonar-scanner-cli.zip.asc /opt/sonar-scanner-cli.zip; \
+    unzip sonar-scanner-cli.zip; \
+    rm sonar-scanner-cli.zip sonar-scanner-cli.zip.asc; \
+    mv sonar-scanner-${SONAR_SCANNER_VERSION} ${SONAR_SCANNER_HOME}; \
+    apk del --purge build-dependencies; \
+    mkdir -p "${SRC_PATH}" "${SONAR_USER_HOME}" "${SONAR_USER_HOME}/cache"; \
+    chown -R scanner-cli:scanner-cli "${SONAR_SCANNER_HOME}" "${SRC_PATH}"; \
+    chmod -R 555 "${SONAR_SCANNER_HOME}"; \
+    chmod -R 754 "${SRC_PATH}" "${SONAR_USER_HOME}";
```

Comment:
Was 777 before, is 754 enough here now?

Reply:
According to the test conducted, I could not find any valid reason why it should be more permissive.